### ISSUE TITLE
How To generate tables in a particular schema as Enums

### ISSUE THREAD

**Issue by hamzus2015** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#482]
Hi, 
I have a schema named ref where I store all reference tables.
I would like to generate Enums for each table.
Exemple:
Table TransportType (id,code,label,description,rank)
Values (1,'URABAIN','Urbain','Urbain Desc') (2,'ROUTE','Route','Route desc')

I need this code to be generated  
Enums TansportType {
URBAIN,
ROUTE
}

Is this scenario possible with Poco reverse generator ?

Thanks

**Comment by sjh37** [comment]
Yes, v4 can do this. For a few tables, list them in `Settings.Enumerations`:

```csharp
Settings.Enumerations = new List<EnumerationSettings>
{
    new EnumerationSettings
    {
        Name       = "TransportType",
        Table      = "ref.TransportType",
        NameField  = "code",
        ValueField = "id"
    }
};
```

To pick up every table in the `ref` schema automatically, use the `Settings.AddEnum` callback instead:

```csharp
Settings.AddEnum = delegate (Table table)
{
    if (table.Schema.DbName != "ref")
        return;

    Settings.Enumerations.Add(new EnumerationSettings
    {
        Name       = table.NameHumanCase,
        Table      = table.Schema.DbName + "." + table.DbName,
        NameField  = "code",
        ValueField = "id"
    });

    table.RemoveTable = true; // optional: generate only the enum, not an entity
};
```

`Settings.ElementsToGenerate` must include both `Elements.Poco` and `Elements.Enum` for `AddEnum` to run. Adding `DescriptionField = "description"` puts a `[Description]` attribute on each member. See [Enum Generation from Table Data](https://github.com/ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator/wiki/Enum-Generation-from-Table-Data).

Closing, as this is supported in v4.
